coda-prototype/src/ui/ColourUtils.ts

```diff
@@ -0,0 +1,182 @@
+/*
+Copyright (c) 2017 Coda authors
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+*/
+
+class ColourUtils {
+    static rgb2hsl(rgbString: string) : string {
+    // based on formulas from http://www.rapidtables.com/convert/color/rgb-to-hsl.htm
+    let rgb = rgbString.split("(")[1].split(")")[0].split(",");
+    let r, g, b, a;
+
+    if (rgb.length == 3) {
+        r = parseInt(rgb[0]);
+        g = parseInt(rgb[1]);
+        b = parseInt(rgb[2]);
+    } else return "";
+
+    if (r >= 0 && 256 > r && g >= 0 && 256 > g && b >= 0 && 256 > b) {
+
+        let r0 = r / 255,
+            g0 = g / 255,
+            b0 = b / 255;
+
+        let min = Math.min(r0, g0, b0),
+            max = Math.max(r0, g0, b0),
+            delta = max - min;
+
+        let switchVal = delta == 0 ? 0 : max;
+        let hue;
+        switch (switchVal) {
+            case 0:
+                hue = 0;
+                break;
+            case r0:
+                hue = 60 * (((g0 - b0) / delta) % 6);
+                break;
+            case g0:
+                hue = 60 * (((b0 - r0) / delta) + 2);
+                break;
+            case b0:
+                hue = 60 * (((r0 - g0) / delta) + 4);
+                break;
+        }
+
+        if (hue >= 360) {
+            hue = hue - 360;
+        } else if (hue < 0) {
+            hue = hue + 360;
+        }
+
+        let luminance = (max + min) / 2;
+        let saturation = delta == 0 ? 0 : (delta / (1 - Math.abs(2 * luminance - 1)));
+
+        return "hsl(" + Math.round(hue * 100) / 100 + "," + Math.round(saturation * 100) / 100 + "," + Math.round(luminance * 100) / 100 + ")";
+
+    } else return "";
+
+    }
+
+    static hex2rgb(hex : string) : string {
```

Comment:
TypeScript style is "arg: type" rather than "arg : type"

Reply:
ack thanks, I'll fix with a global lint pass